src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/ConfiguredProjectLanguageServiceTelemetryService.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    [Export(typeof(IConfiguredProjectLanguageServiceTelemetryService))]
+    internal class ConfiguredProjectLanguageServiceTelemetryService : IConfiguredProjectLanguageServiceTelemetryService
+    {
+        private readonly ConfiguredProject _project;
+        private readonly ITelemetryService _telemetryService;
+        private string? _projectHash;
+
+        [ImportingConstructor]
+        public ConfiguredProjectLanguageServiceTelemetryService(ConfiguredProject project, ITelemetryService telemetryService)
+        {
+            _project = project;
+            _telemetryService = telemetryService;
+        }
+
+        private string ProjectTelemetryId
+        {
+            get
+            {
+                if (Strings.IsNullOrEmpty(_projectHash))
```

Comment:
Should this just be a null check? It can be assigned `""` in this block. I wouldn't expect `project.FullPath` to become non-null at any point. Ditto in other service classes.

This chunk of code is repeated a few times in this PR. Perhaps you could push it into the `HashValue` method (under a slightly different name), making this property something like:

```c#
private string ProjectTelemetryId => _projectHash ??= _telemetryService.GetProjectId(_project?.UnconfiguredProject);
```